[Pages] How to guide on www redirect (#2665)
cloudflare/cloudflare-docs: products/pages/src/content/how-to/www-redirect.md

---
pcx-content-type: how-to
---

# Redirecting www to domain apex

> This tutorial will go over how to redirect the `www` subdomain to your apex domain (`example.com`). This is a common question and can be achieved by using [Page Rules](https://support.cloudflare.com/hc/en-us/articles/218411427).

Reviewer: ```suggestion
In this tutorial, you will learn how to redirect the `www` subdomain to your apex domain (`example.com`). This is a common question and can be achieved by using [Page Rules](https://support.cloudflare.com/hc/en-us/articles/218411427).
```

> To do this, we will need a **proxied** `www` record. We need to create the record so that DNS can look it up and it will need to be proxied so that Cloudflare can handle the redirect.

Reviewer: ```suggestion
To do this, you will need a proxied (orange-clouded) `www` record. You need to create the record so that DNS can look it up and it will need to be proxied so that Cloudflare can handle the redirect.
```

> To do this, you will need a proxied (orange-clouded) `www` record. You need to create the record so that DNS can look it up and it will need to be proxied so that Cloudflare can handle the redirect.

Reviewer: ```suggestion
To do this, create a DNS record for the `www` subdomain. It must either be an `A` record with the `192.0.2.1` value or an `AAAA` record with the `100::` value. The new DNS record must also have Cloudflare's proxy enabled (orange-clouded) so that it can enact the Page Rule behavior that you will create next.
```

> Firstly, create a DNS record for your `www` subdomain. It's recommended that this is either an A or AAAA with the value of `192.0.2.1` or `100::` respectively.

Reviewer: ```suggestion
Firstly, create a DNS record for your `www` subdomain. It is recommended that this is either an A or AAAA with the value of `192.0.2.1` or `100::` respectively.
```

> Firstly, create a DNS record for your `www` subdomain. It is recommended that this is either an A or AAAA with the value of `192.0.2.1` or `100::` respectively.

Reviewer: ```suggestion
```

> 

Reviewer: ```suggestion
```

![www record](./media/www_subdomain.png)

> Now we can make the Page Rule which will handle the redirect. For this, we will make it so all paths get redirected and it's done with a 301 (Permanent Redirect). Matching all paths allows for someone visiting `www.example.com/blog/category/example` to be redirected to `example.com/blog/category/example`.

Reviewer: ```suggestion
Next, make the Page Rule which will handle the redirect. For this, you will make it so all paths get redirected and it is done with a 301 (Permanent Redirect). Matching all paths allows for someone visiting `www.example.com/blog/category/example` to be redirected to `example.com/blog/category/example`.
```

> To do this, navigate to [Page Rules in the dashboard](https://dash.cloudflare.com?to=/:account/:zone/rules) and click "Create Page Rule". Then set the target to `www.example.com/*`, the `/*` is what matches all paths. Pick "Forwarding URL" from the dropdown and select the "301 - Permanent Redirect" option from the status code dropdown. Finally, for the Forwarding URL location, we want to set it to `example.com/$1`, the `$1` will be replaced by our wildcard match in target (the `*`).

Reviewer: ```suggestion
To do this:
1. Go to **Rules** [in the dashboard](https://dash.cloudflare.com?to=/:account/:zone/rules).
2. Select **Create Page Rule**. 
3. Set the target to `www.example.com/*`, the `/*` is what matches all paths. 
4. Select *Forwarding URL* from the dropdown and select the **301 - Permanent Redirect** option from the status code dropdown. 
5. Finally, for the Forwarding URL location, set it to `example.com/$1`, the `$1` will be replaced by our wildcard match in target (the `*`).
```

Author: Any reason why "Forwarding URL" is italic rather than bold?

Reviewer: @WalshyDev hi! it's a small Cloudflare style guide difference but bolds are for anything that is select/clickable (tabs, buttons, sections, etc.) but italics are used exclusively for when someone has to pick from a dropdown list of options.

Author: Ah interesting! Thank you Kate :)

![Page Rule redirect](./media/www_redirect_pagerule.png)

> Finally, click "Save" and wait a few seconds for your rule to propagate!

Reviewer: ```suggestion
Finally, select **Save** and wait a few seconds for your rule to propagate.
```
